src/midje/emission/plugins/util.clj

```diff
@@ -141,15 +141,22 @@
         : a nicely printed stack trace
         : maps and sets sorted by key."
   [v]
-  (let [raw-str (cond
-                  (fn? v)                           (function-name v)
-                  (exception/captured-throwable? v) (exception/friendly-stacktrace v)
-                  (record? v)                       (str (sorted-if-appropriate v) "::" (record-name v))
-                  :else                             (sorted-if-appropriate v))]
-    (puget/cprint-str raw-str {:print-handlers {Metaconstant puget/pr-handler}
-                               :print-fallback :pretty
-                               :seq-limit      10
-                               :map-delimiter  ""})))
+  (if (exception/captured-throwable? v)
+    (exception/friendly-stacktrace v)
+    (let [raw-str (cond (fn? v)     (function-name v)
+                        (record? v) (str (sorted-if-appropriate v) "::" (record-name v))
+                        :else       (sorted-if-appropriate v))]
+      (cond
+        (config/choice :pretty-print)
+        (puget/cprint-str raw-str {:print-handlers {Metaconstant puget/pr-handler}
+                                   :print-fallback :pretty
+                                   :seq-limit      10
+                                   :map-delimiter  ""})
+        (string? raw-str)
+        raw-str
+
+        :else
+        (pr-str raw-str)))))
```

Comment:
These last two conditions are needed because before we turned the `(sorted-if-appropriate ..)` into a string https://github.com/marick/Midje/pull/403/files#diff-e3c818e11b1219e71f741f998579a6faR148, but now we don't in case we pretty-print it, so if we don't use pretty-printing, we need to do the string conversion again